ms/MSOper/MSConcat.cc

```diff
@@ -668,7 +672,27 @@ IPosition MSConcat::isFixedShape(const TableDesc& td) {
     }  
   }
 
+  const ROMSPolarizationColumns otherPolCols(otherMS.polarization());
+  const ROMSDataDescColumns otherDDCols(otherMS.dataDescription());
+  Int polId = -1;
+  vector<Int> polSwap;
+
   for (uInt r = 0; r < otherRows; r++) {
+    uInt d = otherDDId(r);
+    uInt p = otherDDCols.polarizationId()(otherDDId(r));
+    if (p != polId) {
+      const Matrix<Int> &products = otherPolCols.corrProduct()(p);
+      polSwap.resize(products.shape()(1));
+      for (Int i = 0; i < products.shape()(1); i++) {
+	for (Int j = 0; j < products.shape()(1); j++)
```

Comment:
The second "for" should be indented.

Reply:
And it should have a pair of brackets!

Reply:
In fact, could you please add a comment to the code to explain what exactly is supposed to happen here?
I'm not sure I understand.